app/services/supplementary_data.py

```diff
@@ -30,7 +33,20 @@ def __str__(self) -> str:
         return "Supplementary Data request failed"
 
 
+class MissingSupplementaryDataKey(Exception):
+    def __str__(self) -> str:
+        return "Missing supplementary data key"
+
+
+class InvalidSupplementaryData(Exception):
+    def __str__(self) -> str:
+        return "Supplementary data invalid"
+
+
 def get_supplementary_data(*, dataset_id: str, unit_id: str, survey_id: str) -> dict:
+    if not get_key_store().get_key(purpose=KEY_PURPOSE_SDS, key_type="private"):
+        raise MissingSupplementaryDataKey()
```

Comment:
Non blocker, if you are not passing anything into the exception, you don't need to instantiate it. https://docs.python.org/3/tutorial/errors.html#raising-exceptions
```suggestion
        raise MissingSupplementaryDataKey
```